src/ChunkMap.cpp

```diff
@@ -1313,21 +1313,14 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 				a_Entity.TakeDamage(dtExplosion, nullptr, static_cast<int>((1 / std::max(1.0, DistanceFromExplosion.Length())) * 8 * ExplosionSizeInt), 0);
 			}
 
-			double Length = DistanceFromExplosion.Length();
-			if (Length <= ExplosionSizeInt)  // Entity is impacted by explosion
-			{
-				float EntityExposure = a_Entity.GetExplosionExposureRate(ExplosionPos, static_cast<float>(a_ExplosionSize));
-
-				// Exposure reduced by armor
-				EntityExposure = EntityExposure * (1.0f - a_Entity.GetEnchantmentBlastKnockbackReduction());
+			// Hotfix for #4575
+			// TODO: Enhance knockback to be more vanilla-like. Reapply blast reduction.
+			auto Knockback = (1 / std::max(1.0, DistanceFromExplosion.Length())) * 6 * ExplosionSizeInt;
 
-				double Impact = (1 - ((Length / ExplosionSizeInt) / 2)) * EntityExposure;
+			DistanceFromExplosion.Normalize();
+			DistanceFromExplosion = Vector3d(DistanceFromExplosion.x * Knockback, DistanceFromExplosion.y + (Knockback * 0.3), DistanceFromExplosion.z * Knockback);
```

Comment:
Vector3s support vector multiplication, so no need to decompose and recompose.

```suggestion
			DistanceFromExplosion *= Vector3d(1, 0.3, 1) * Knockback;
```

Reply:
This breaks height knockback (notice the +).